CI build fails for tests due to thread function signature changes

Build failures:

    ccache /opt/toolchains/zephyr-sdk-0.16.3/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-gcc -DKERNEL -DPICOLIBC_INTEGER_PRINTF_SCANF -DTC_RUNID=aa27f2621db6b33dd47811b2391bc39b -D_FORTIFY_SOURCE=1 -D_POSIX_C_SOURCE=200809 -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I/__w/zephyr/zephyr/include -I/__w/zephyr/zephyr/twister-out/qemu_x86_64/tests/kernel/mem_protect/stackprot/kernel.memory_protection.stackprot_tls/zephyr/include/generated -I/__w/zephyr/zephyr/soc/x86/ia32 -I/__w/zephyr/zephyr/subsys/testsuite/include -I/__w/zephyr/zephyr/subsys/testsuite/ztest/include -m64  -fno-strict-aliasing -Werror -fstack-protector-all -mstack-protector-guard=tls -Os -imacros /__w/zephyr/zephyr/twister-out/qemu_x86_64/tests/kernel/mem_protect/stackprot/kernel.memory_protection.stackprot_tls/zephyr/include/generated/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -m64 -Wa,--divide --sysroot=/opt/toolchains/zephyr-sdk-0.16.3/x86_64-zephyr-elf/x86_64-zephyr-elf -imacros /__w/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/tests/kernel/mem_protect/stackprot=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -mno-sse3 -mno-ssse3 -mno-sse4.1 -mno-sse4.2 -mno-sse4a --specs=picolibc.specs -std=c99 -MD -MT CMakeFiles/app.dir/src/main.c.obj -MF CMakeFiles/app.dir/src/main.c.obj.d -o CMakeFiles/app.dir/src/main.c.obj -c /__w/zephyr/zephyr/tests/kernel/mem_protect/stackprot/src/main.c
    /__w/zephyr/zephyr/tests/kernel/mem_protect/stackprot/src/main.c: In function 'stackprot_test_create_alt_thread':
    /__w/zephyr/zephyr/tests/kernel/mem_protect/stackprot/src/main.c:132:25: error: passing argument 4 of 'k_thread_create' from incompatible pointer type [-Werror=incompatible-pointer-types]
      132 |                         alternate_thread, NULL, NULL, NULL,
          |                         ^~~~~~~~~~~~~~~~
          |                         |
          |                         void (*)(void)
    In file included from /__w/zephyr/zephyr/include/zephyr/kernel.h:6102,
                     from /__w/zephyr/zephyr/tests/kernel/mem_protect/stackprot/src/main.c:9:
    /__w/zephyr/zephyr/twister-out/qemu_x86_64/tests/kernel/mem_protect/stackprot/kernel.memory_protection.stackprot_tls/zephyr/include/generated/syscalls/kernel.h:59:131: note: expected 'k_thread_entry_t' {aka 'void (*)(void *, void *, void *)'} but argument is of type 'void (*)(void)'
       59 | static inline k_tid_t k_thread_create(struct k_thread * new_thread, k_thread_stack_t * stack, size_t stack_size, k_thread_entry_t entry, void * p1, void * p2, void * p3, int prio, uint32_t options, k_timeout_t delay)
          |                                                                                                                  ~~~~~~~~~~~~~~~~~^~~~~
    cc1: all warnings being treated as errors
    
    /__w/zephyr/zephyr/tests/kernel/mutex/sys_mutex/src/main.c:91:25: error: passing argument 4 of 'k_thread_create' from incompatible pointer type [-Werror=incompatible-pointer-types]
       91 |                         thread_##id,                                                       \
          |                         ^~~~~~~
          |                         |
          |                         void (*)(void)
    /__w/zephyr/zephyr/tests/kernel/mutex/sys_mutex/src/main.c:269:9: note: in expansion of macro 'CREATE_PARTICIPANT_THREAD'
      269 |         CREATE_PARTICIPANT_THREAD(05, 5);
          |         ^~~~~~~~~~~~~~~~~~~~~~~~~
    In file included from /__w/zephyr/zephyr/include/zephyr/kernel.h:6102,
                     from /__w/zephyr/zephyr/subsys/testsuite/include/zephyr/tc_util.h:12,
                     from /__w/zephyr/zephyr/tests/kernel/mutex/sys_mutex/src/main.c:48:
    
    /__w/zephyr/zephyr/tests/kernel/sched/metairq/src/main.c:155:70: error: passing argument 4 of 'k_thread_create' from incompatible pointer type [-Werror=incompatible-pointer-types]
      155 |         CREATE_PARTICIPANT_THREAD(metairq_thread_id, K_PRIO_COOP(0), metairq_thread);
          |                                                                      ^~~~~~~~~~~~~~
          |                                                                      |
          |                                                                      void (*)(void)
    /__w/zephyr/zephyr/tests/kernel/sched/metairq/src/main.c:43:25: note: in definition of macro 'CREATE_PARTICIPANT_THREAD'
       43 |                         entry,                                                         \
          |                         ^~~~~
    In file included from /__w/zephyr/zephyr/include/zephyr/kernel.h:6102,
                     from /__w/zephyr/zephyr/tests/kernel/sched/metairq/src/main.c:6:
    /__w/zephyr/zephyr/twister-out/qemu_x86_64/tests/kernel/sched/metairq/kernel.scheduler.metairq/zephyr/include/generated/syscalls/kernel.h:59:131: note: expected 'k_thread_entry_t' {aka 'void (*)(void *, void *, void *)'} but argument is of type 'void (*)(void)'
       59 | static inline k_tid_t k_thread_create(struct k_thread * new_thread, k_thread_stack_t * stack, size_t stack_size, k_thread_entry_t entry, void * p1, void * p2, void * p3, int prio, uint32_t options, k_timeout_t delay)
          |                                                                                                                  ~~~~~~~~~~~~~~~~~^~~~~
    /__w/zephyr/zephyr/tests/kernel/sched/metairq/src/main.c:156:68: error: passing argument 4 of 'k_thread_create' from incompatible pointer type [-Werror=incompatible-pointer-types]
      156 |         CREATE_PARTICIPANT_THREAD(coop_thread1_id, K_PRIO_COOP(1), coop_thread1);
          |                                                                    ^~~~~~~~~~~~
          |                                                                    |
          |                                                                    void (*)(void)
    /__w/zephyr/zephyr/tests/kernel/sched/metairq/src/main.c:43:25: note: in definition of macro 'CREATE_PARTICIPANT_THREAD'
       43 |                         entry,                                                         \
          |                         ^~~~~
    /__w/zephyr/zephyr/twister-out/qemu_x86_64/tests/kernel/sched/metairq/kernel.scheduler.metairq/zephyr/include/generated/syscalls/kernel.h:59:131: note: expected 'k_thread_entry_t' {aka 'void (*)(void *, void *, void *)'} but argument is of type 'void (*)(void)'
       59 | static inline k_tid_t k_thread_create(struct k_thread * new_thread, k_thread_stack_t * stack, size_t stack_size, k_thread_entry_t entry, void * p1, void * p2, void * p3, int prio, uint32_t options, k_timeout_t delay)
          |                                                                                                                  ~~~~~~~~~~~~~~~~~^~~~~
    /__w/zephyr/zephyr/tests/kernel/sched/metairq/src/main.c:157:68: error: passing argument 4 of 'k_thread_create' from incompatible pointer type [-Werror=incompatible-pointer-types]
      157 |         CREATE_PARTICIPANT_THREAD(coop_thread2_id, K_PRIO_COOP(2), coop_thread2);
          |                                                                    ^~~~~~~~~~~~
          |                                                                    |
          |                                                                    void (*)(void)
    /__w/zephyr/zephyr/tests/kernel/sched/metairq/src/main.c:43:25: note: in definition of macro 'CREATE_PARTICIPANT_THREAD'
       43 |                         entry,                                                         \
          |                         ^~~~~
    
This was introduced by PR #62903

--- a/tests/kernel/mem_protect/stackprot/src/main.c
+++ b/tests/kernel/mem_protect/stackprot/src/main.c
@@ -78,8 +78,12 @@ void __attribute__((noinline)) check_input(const char *name, const char *input)
  * and will not set ret to TC_FAIL.
  *
  */
-void alternate_thread(void)
+void alternate_thread(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
 	TC_PRINT("Starts %s\n", __func__);
 	check_input(__func__,
 		    "Input string is too long and stack overflowed!\n");

--- a/tests/kernel/mutex/sys_mutex/src/main.c
+++ b/tests/kernel/mutex/sys_mutex/src/main.c
@@ -100,8 +100,12 @@ static ZTEST_BMEM SYS_MUTEX_DEFINE(bad_count_mutex);
  *
  */
 
-void thread_05(void)
+void thread_05(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
 	int rv;
 
 	k_sleep(K_MSEC(3500));
@@ -122,8 +126,12 @@ void thread_05(void)
  *
  */
 
-void thread_06(void)
+void thread_06(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
 	int rv;
 
 	k_sleep(K_MSEC(3750));
@@ -153,8 +161,12 @@ void thread_06(void)
  *
  */
 
-void thread_07(void)
+void thread_07(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
 	int rv;
 
 	k_sleep(K_MSEC(2500));
@@ -182,8 +194,12 @@ void thread_07(void)
  *
  */
 
-void thread_08(void)
+void thread_08(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
 	int rv;
 
 	k_sleep(K_MSEC(1500));
@@ -205,8 +221,12 @@ void thread_08(void)
  *
  */
 
-void thread_09(void)
+void thread_09(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
 	int rv;
 
 	k_sleep(K_MSEC(500));	/* Allow lower priority thread to run */
@@ -237,8 +257,12 @@ void thread_09(void)
  *
  */
 
-void thread_11(void)
+void thread_11(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
 	int rv;
 
 	k_sleep(K_MSEC(3500));
@@ -253,7 +277,7 @@ void thread_11(void)
 
 K_THREAD_STACK_DEFINE(thread_12_stack_area, STACKSIZE);
 struct k_thread thread_12_thread_data;
-extern void thread_12(void);
+extern void thread_12(void *p1, void *p2, void *p3);
 
 
 

--- a/tests/kernel/sched/metairq/src/main.c
+++ b/tests/kernel/sched/metairq/src/main.c
@@ -59,8 +59,12 @@ volatile int coop_cnt2;
 #define LOOP_CNT  4 /* Number of times low priority thread waits */
 
 /* Meta-IRQ thread */
-void metairq_thread(void)
+void metairq_thread(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
 	k_sem_take(&metairq_sem, K_FOREVER);
 
 	printk("metairq start\n");
@@ -82,8 +86,12 @@ void metairq_thread(void)
 }
 
 /* High-priority cooperative thread */
-void coop_thread1(void)
+void coop_thread1(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
 	int cnt1, cnt2;
 
 	printk("thread1 take sem\n");
@@ -109,8 +117,12 @@ void coop_thread1(void)
 }
 
 /* Low-priority cooperative thread */
-void coop_thread2(void)
+void coop_thread2(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
 	int cnt1, cnt2;
 
 	printk("thread2 take sem\n");

--- a/tests/kernel/sleep/src/main.c
+++ b/tests/kernel/sleep/src/main.c
@@ -102,10 +102,10 @@ static int sleep_time_valid(uint32_t start, uint32_t end, uint32_t dur)
 
 static void test_thread(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
 	ARG_UNUSED(p3);
 
-	int arg1 = POINTER_TO_INT(p1);
-	int arg2 = POINTER_TO_INT(p2);
 	uint32_t start_tick;
 	uint32_t end_tick;
 
@@ -177,10 +177,9 @@ static void irq_offload_isr(const void *arg)
 
 static void helper_thread(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
 	ARG_UNUSED(p3);
-
-	int arg1 = POINTER_TO_INT(p1);
-	int arg2 = POINTER_TO_INT(p2);
 
 	k_sem_take(&helper_thread_sem, K_FOREVER);
 	/* Wake the test thread */

--- a/tests/kernel/smp/src/main.c
+++ b/tests/kernel/smp/src/main.c
@@ -1079,15 +1079,15 @@ static int run_concurrency(void *p1, void *p2, void *p3)
 ZTEST(smp, test_inc_concurrency)
 {
 	/* increasing global var with irq lock */
-	zassert_true(run_concurrency(LOCK_IRQ, inc_global_cnt),
+	zassert_true(run_concurrency(INT_TO_POINTER(LOCK_IRQ), inc_global_cnt, NULL),
 			"total count %d is wrong(i)", global_cnt);
 
 	/* increasing global var with irq lock */
-	zassert_true(run_concurrency(LOCK_SEM, inc_global_cnt),
+	zassert_true(run_concurrency(INT_TO_POINTER(LOCK_SEM), inc_global_cnt, NULL),
 			"total count %d is wrong(s)", global_cnt);
 
 	/* increasing global var with irq lock */
-	zassert_true(run_concurrency(LOCK_MUTEX, inc_global_cnt),
+	zassert_true(run_concurrency(INT_TO_POINTER(LOCK_MUTEX), inc_global_cnt, NULL),
 			"total count %d is wrong(M)", global_cnt);
 }
 

